Parse license in kiwi image packages list

aaannz opened the pull request:
## What does this PR change?
Kiwi-NG in upcomming update added package license text at the end of image package list.
For backward compatibility the license match is optional, but the rest patterns were modified to be non-greedy.

## GUI diff

No difference.

- [X] **DONE**

## Documentation
- No documentation needed: internal data parsing

- [X] **DONE**

## Test coverage
- Unit tests were updated
- Already part of the cucumber SLE15 image building test scenario

- [X] **DONE**

## Links

Backport https://github.com/SUSE/spacewalk/pull/10118

- [x] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)


## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"   
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_lint_checkstyle"		 
- [ ] Re-run test "java_pgsql_tests"		 
- [ ] Re-run test "ruby_rubocop"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"		 
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"		 
- [ ] Re-run test "spacecmd_unittests"

aaannz commented:
The changelog test fails because I altered inspection test json for unit tests and didn't add entry to spacewalk-java.changes. I don't think that the changelog entry is needed just for unit tests data change.
The actual change has changelog entry in susemanager-sls.changes.

nadvornik commented:
I think we should test both variants, with and without license.

mbologna commented:
> Tracks # **add downstream PR, if any**
> 
>     *

Do you plan to backport this PR to Manager* branches?

mbologna commented:
> The changelog test fails because I altered inspection test json for unit tests and didn't add entry to spacewalk-java.changes. I don't think that the changelog entry is needed just for unit tests data change.
> The actual change has changelog entry in susemanager-sls.changes.

I think you will need to add a changelog for `spacewalk-java` too. IIRC The reason is: when we are going to build the next version of the package and the build mechanism will see a change without changelog it will complain. And you will be asked to provide a changelog.

juliogonzalez commented:
@mbologna is right, in this specific case `spacewalk-java` needs the changelog entry we have at `susemanager-sls` or there could be questions in the future.

aaannz commented:
Ok, I'll add the changelog entry. I also plan to backport it to 4.0, I'll do the changes after I finish my tests with SLE12 images.

aaannz commented:
Added backport - https://github.com/SUSE/spacewalk/pull/10118
Added changelog entry, but seems that the failed changelog tests are not being updated.

mbologna commented:
> Added backport - [SUSE/spacewalk#10118](https://github.com/SUSE/spacewalk/pull/10118)
> Added changelog entry, but seems that the failed changelog tests are not being updated.

Yes, as a temporary problem, changelogs are failing (see R.C). Feel free to merge at your earliest convenience.
Thanks!

aaannz commented:
Unfortunately I can't merge even when I tried changelog is not required checkbox. Can someone override and merge this for me?
